File: frontend/src/types/host.ts
export interface HostSummary {
  id: number;
  name: string;
  address: string;
  port: number;
  status: string;
  agent_version: string;
  capabilities?: string[];
  last_seen_at?: string | null;
  enrollment_id?: string;
  revoked: boolean;
  revoked_at?: string | null;
  revoked_reason?: string;
  created_at: string;
  updated_at: string;
}

export interface ListHostsResult {
  items: HostSummary[];
}

export interface HostFormInput {
  name?: string;
  address: string;
  port: number;
}

export interface EnrollHostInput {
  token: string;
  hostname: string;
  agent_version?: string;
  capabilities?: string[];
}

export interface RotateCertificateInput {
  reuse_key: boolean;
}

File: backend/internal/dto/host.go
package dto

type CreateHostRequest struct {
	Name    string `json:"name"`
	Address string `json:"address" binding:"required"`
	Port    int    `json:"port" binding:"required"`
}

type UpdateHostRequest struct {
	Name    string `json:"name"`
	Address string `json:"address" binding:"required"`
	Port    int    `json:"port" binding:"required"`
}

type EnrollHostRequest struct {
	Token        string   `json:"token" binding:"required"`
	Hostname     string   `json:"hostname" binding:"required"`
	AgentVersion string   `json:"agent_version"`
	Capabilities []string `json:"capabilities"`
}

type Host struct {
	ID              int64    `json:"id"`
	Name            string   `json:"name"`
	Address         string   `json:"address"`
	Port            int      `json:"port"`
	Status          string   `json:"status"`
	AgentVersion    string   `json:"agent_version"`
	Capabilities    []string `json:"capabilities,omitempty"`
	LastSeenAt      *string  `json:"last_seen_at,omitempty"`
	EnrollmentID    string   `json:"enrollment_id,omitempty"`
	Revoked         bool     `json:"revoked"`
	RevokedAt       *string  `json:"revoked_at,omitempty"`
	RevokedReason   string   `json:"revoked_reason,omitempty"`
	CreatedAt       string   `json:"created_at"`
	UpdatedAt       string   `json:"updated_at"`
}

type ListHostsResult struct {
	Items []Host `json:"items"`
}
